src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -4,32 +4,47 @@
 using System;
 using System.ComponentModel;
 using System.Reflection;
-
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 namespace Microsoft.Azure.SignalR.Management
 {
     /// <summary>
     /// A builder for configuring <see cref="IServiceManager"/> instances.
     /// </summary>
-    public class ServiceManagerBuilder : IServiceManagerBuilder
+    public class ServiceManagerBuilder : IServiceManagerBuilder, IDisposable
     {
-        private readonly ServiceManagerOptions _options = new ServiceManagerOptions();
+        private readonly IServiceCollection _services = new ServiceCollection();
         private Assembly _assembly;
+        private ServiceProvider _serviceProvider;
 
         /// <summary>
-        /// Configures the <see cref="IServiceManager"/> instances.
+        /// Registers an action used to configure <see cref="IServiceManager"/>.
         /// </summary>
         /// <param name="configure">A callback to configure the <see cref="IServiceManager"/>.</param>
         /// <returns>The same instance of the <see cref="ServiceManagerBuilder"/> for chaining.</returns>
         public ServiceManagerBuilder WithOptions(Action<ServiceManagerOptions> configure)
         {
-            configure?.Invoke(_options);
+            _services.Configure(configure);
+            return this;
+        }
+
+        /// <summary>
+        /// Registers a configuration instance to configure <see cref="IServiceManager"/>
+        /// </summary>
+        /// <param name="config">The configuration instance.</param>
+        /// <returns>The same instance of the <see cref="ServiceManagerBuilder"/> for chaining.</returns>
+        internal ServiceManagerBuilder WithConfiguration(IConfiguration config)
+        {
+            _services.Configure<ServiceManagerOptions>(config.GetSection(Constants.Keys.ServiceManagerOptionsSectionKey));
```

Comment:
if customer uses `ServiceManagerBuilder`, does it support hot reload?

Reply:
Yes, the test `ConfigHotReloadTest` of `ServiceManagerBuilder` tests the hot reload .